aioauth/response_type.py

```diff
@@ -125,14 +125,15 @@ async def create_authorization_response(
         self, request: TRequest, client: Client
     ) -> AuthorizationCodeResponse:
         authorization_code = await self.storage.create_authorization_code(
-            request,
-            client.client_id,
-            request.query.scope,
-            request.query.response_type,  # type: ignore
-            request.query.redirect_uri,
-            request.query.code_challenge_method,
-            request.query.code_challenge,
-            generate_token(42),
+            client_id=client.client_id,
```

Comment:
I was thinking of allowing only positional arguments, in which case there will be an explicit indication of what arguments are required:

```python
    async def create_authorization_code(
        self,
        request: TRequest,
        *,
        client_id: str,
        scope: str,
        response_type: ResponseType,
        redirect_uri: str,
        code_challenge_method: Optional[CodeChallengeMethod],
        code_challenge: Optional[str],
        code: str,
    ) -> TAuthorizationCode:
```

it's should not be done in scope of this PR, just as an idea. What do you think?

Reply:
I'm flexible on the matter. I tend to feel like positional arguments can get unwieldy when there is a lot of them and I also like that keyword arguments help someone reading the code outside of an IDE (e.g. in Github for example) to better understand what the purpose of a given argument is without having to go look at the function that is being called to understand what it's positional arguments are.

To that end, I've updated the signature for `create_authorization_code` and `get_id_token` to include a tail `**kwargs` to allow for keyword args to be passed that may be used by a subclass of `Storage`. I think this achieves kind of the same thing in that the required arguments are still explicit, but other implementations can take additional arguments as needed.